hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextQueryProcessor.java

```diff
@@ -72,18 +74,30 @@ public ContextQueryProcessor(
 
     @Override
     public CallOutcome call() {
-        // Try to translate the HAPI operation to a Hedera EVM transaction, throw HandleException on failure
-        final var hevmTransaction = hevmStaticTransactionFactory.fromHapiQuery(context.query());
+        try {
+            // Try to translate the HAPI operation to a Hedera EVM transaction, throw HandleException on failure
+            final var hevmTransaction = hevmStaticTransactionFactory.fromHapiQuery(context.query());
 
-        final var contractsConfig = context.configuration().getConfigData(ContractsConfig.class);
-        // Get the appropriate processor for the EVM version
-        final var processor = processors.get(EVM_VERSIONS.get(contractsConfig.evmVersion()));
+            final var contractsConfig = context.configuration().getConfigData(ContractsConfig.class);
+            // Get the appropriate processor for the EVM version
+            final var processor = processors.get(EVM_VERSIONS.get(contractsConfig.evmVersion()));
 
-        // Process the transaction
-        final var result = processor.processTransaction(
-                hevmTransaction, worldUpdater, feesOnlyUpdater, hederaEvmContext, tracer, context.configuration());
+            // Process the transaction
+            final var result = processor.processTransaction(
+                    hevmTransaction, worldUpdater, feesOnlyUpdater, hederaEvmContext, tracer, context.configuration());
 
-        // Return the outcome (which cannot include sidecars to be externalized, since this is a query)
-        return CallOutcome.fromResultsWithoutSidecars(result.asQueryResult(), result);
+            // Return the outcome (which cannot include sidecars to be externalized, since this is a query)
+            return CallOutcome.fromResultsWithoutSidecars(result.asQueryResult(), result);
+        } catch (final HandleException e) {
+            final var op = context.query().contractCallLocalOrThrow();
+            final var senderId = op.hasSenderId() ? op.senderIdOrThrow() : context.payer();
+
+            final var hevmTransaction = hevmStaticTransactionFactory.fromHapiQueryException(context.query(), e);
```

Comment:
I'm wondering if you actually need to create the `HederaEvmTransaction`?  You can get the contract id via `context.query().contractCallLocalOrThrow().contractIdOrThrow()` _I think_ and you have the exception in hand.